### PR TITLE
Feature/form functionality and validation

### PR DESCRIPTION
**Pull request by lschnell8**
Is this a fix or a feature?
- [ ] Bug fix
- [x] Feature
- [x] Refactor
 
Describe the feature - fix - refactor:
Set up form validation and error handling for donate and request milk forms. Cleaned up unused code and unified syntax style in all files.

How was this tested?
- [ ] open index.html
- [x] localhost
- [ ] terminal

Where should the reviewer start - test:
For error handling functionality DonateMilkForm.js and RequestMilkForm.js 

Additional Comments: 
I resolved merge conflicts on this branch but looks like something went wrong with Heroku deployment. The error in the build is `failed to push some refs to 'https://git.heroku.com/******************.git`. One of the hints said to pull down and push again. I pulled, resolved conflicts locally, and tried to push again but my push was rejected. I feel like we've had this error before, but I couldn't quite remember. If I could get some more eyes on it, I'd appreciate it.

### CODE REVIEW

File: src/Components/NeedMilk/NeedMilk.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import DonorCard from '../DonorCard/DonorCard';
-import {useEffect, useState} from 'react';
-import {useQuery, gql} from '@apollo/client';
+import { useEffect, useState } from 'react';
+import { useQuery } from '@apollo/client';
 import { GET_ALL_DONORS } from '../../Graphql/Queries';
 
 const NeedMilk = () => {
-  const {error, loading, data} = useQuery(GET_ALL_DONORS)
```
Comment (lschnell8) [inline src/Components/NeedMilk/NeedMilk.js:7]: I took these out to clean up files, but do we need them for error handling components?

File: package.json [hunk 2]
```diff
@@ -12,6 +12,7 @@
     "react": "^17.0.2",
     "react-burger-menu": "^3.0.6",
     "react-dom": "^17.0.2",
+    "react-hook-form": "^7.28.1",
```
Comment (cagallo) [inline package.json:15]: What a great package!

File: src/Components/DonateMilkForm/DonateMilkForm.js [hunk 3]
```diff
@@ -1,38 +1,34 @@
...
-
-  // const [user, setUser] = useState({
-  //   name: '',
-  //   email: '',
-  //   phone: '',
-  //   bio: '', 
-  //   location: '',
-  //   donor_status: smoker
-  // })
+  const navigate = useNavigate();
```
Comment (cagallo) [inline src/Components/DonateMilkForm/DonateMilkForm.js:8]: Great job cleaning up this file!!

**Comment by janitastic** [comment]
Closing this without a merge to fix circle ci issues.

**Pull request closed without merging**